### PR TITLE
Add native/firstParty flag to collector exporters

### PR DESCRIPTION
**Pull request by haidong**
To address #4795, this is a series of PRs that add native or first party flags to registries that lack them. This batch touches collector-builder and all collector-exporter-* registries.

### CODE REVIEW

**Review by cartermp — APPROVED** [review]

File: data/registry/collector-builder.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ urls:
   repo: https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder
   docs: /docs/collector/custom-collector/
 createdAt: 2023-12-18
+isFirstParty: true
```
Comment (svrnm) [inline data/registry/collector-builder.yml:17]: Apologies, I should have been more specific in the original issue. This flag is not relevant for collector components it refers to instrumentations that are "first party", i.e. provided by the same org that created a certain app or library. 

We could consider to have them for receivers/exporters hosted outside the community repos by orgs that create such modules fur their solution but I don't know if this is what we want them to do va adopting otlp

**Review by svrnm — CHANGES_REQUESTED** [review]
Please see my inline comment. IsFirstParty should not be used for collector components.

File: data/registry/collector-builder.yml (hunk 1 above)
Comment (haidong) [reply data/registry/collector-builder.yml]: I was unsure about this change. Thanks for the explanation.

I've made the adjustment to your change request, rebased and adjusted the commit message a bit, then did a force push.

Thanks again!

Comment (haidong) [reply data/registry/collector-builder.yml]: Just to clarify, Do you mean that _all_ collector-exporters don't need this change? In other words, this PR should be abandoned?

Comment (svrnm) [reply data/registry/collector-builder.yml]: Depends on how we want to use that flag. We had a thirdParty flag in the past that we removed because it created so much confusion. Right now the first party flag is used in the registry to say what I outlined above. I don't know if we water down the value of that if all otel first party components have that label as well

File: layouts/partials/ecosystem/registry/entry.html
Comment (svrnm) [inline layouts/partials/ecosystem/registry/entry.html:1]: Oh, thanks, typo by me! Can you move this in a separate PR since the other topic will need some time to be discussed?

**Comment by svrnm** [comment]
Moving the discussion out from my outdated comment:

> Depends on how we want to use that flag. We had a thirdParty flag in the past that we removed because it created so much confusion. Right now the first party flag is used in the registry to say what I outlined above. I don't know if we water down the value of that if all otel first party components have that label as well

I don't think that we should add the isFirstParty flag to non-instrumentation entries of the registry, especially not for things build by the opentelemetry community. Otherwise 50%+ of our entries would have this flag (it is currently not showing up because `layouts/partials/ecosystem/registry/entry.html` is implementing that flag exactly that way that it only works for instrumentation)

We already highlight components by the opentelemetry community by distinguishing the "by The OpenTelemetry Authors" from other others by highlighting them.

Again, @haidong, I appreciate the effort you put into creating this PR, and I apologies for the confusion that my comment in #4795 might have created, but I would like to not implement this.

**Comment by cartermp** [comment]
Agreed, let's keep this scoped to just instrumentations then.

**Comment by haidong** [comment]
I've been busy with other stuff and didn't get a chance to attend to this until today.

Thanks all for your input. I've created a new PR that touch files under data/registry. Please let me know your suggestions/comments there.
